layout/FallbackQuery.js

```diff
@@ -353,40 +353,36 @@ Layout.prototype.render = function( vs ){
   var q = Layout.base( vs );
```

Comment:
might be cleaner to make another variable here for storing that deep-nested filtered query bool object since it's used throughout the rest of the file and is pretty verbose, like 

```
var funcScoreShould = q.query.function_score.query.filtered.query.bool.should;
...
funcScoreShould.push(...);
```